### ISSUE TITLE
ObjectDecetion custom trained model recognizes objects with many wrong results

### ISSUE THREAD

**Issue by alanjuster** [tensorflow/models#7907]
Please go to Stack Overflow for help and support:

http://stackoverflow.com/questions/tagged/tensorflow

Also, please understand that many of the models included in this repository are experimental and research-style code. If you open a GitHub issue, here is our policy:

1. It must be a bug, a feature request, or a significant problem with documentation (for small docs fixes please send a PR instead).
2. The form below must be filled out.

**Here's why we have that policy**: TensorFlow developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

------------------------

### System information
- **What is the top-level directory of the model you are using**: object_decetion/legacy/train.py
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:  no 
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:  Ubuntu 16.04
- **TensorFlow installed from (source or binary)**:  source
- **TensorFlow version (use command below)**:   tensorflow-gpu  1.15.0
- **Bazel version (if compiling from source)**:   0.26.1
- **CUDA/cuDNN version**:  10.1
- **GPU model and memory**: GTX 1660   6G
- **Exact command to reproduce**:  n/a

### Describe the problem
i use ssd_mobilenet_v1_coco  train my own data( Hammer  ， screwdriver ，Pipe wrenches )
use 2 gpu，  batch_size：30，  train steps  150w，then  save as model ,    This model can identify these three tools, but which other objects are also recognized as tools and the score is still high. What is wrong with this? Thank you.

### Source code / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.
my ssd_mobilenet_v1_coco.config
` 

model {
  ssd {
    num_classes: 3
    box_coder {
      faster_rcnn_box_coder {
        y_scale: 10.0
        x_scale: 10.0
        height_scale: 5.0
        width_scale: 5.0
      }
    }
    matcher {
      argmax_matcher {
        matched_threshold: 0.5
        unmatched_threshold: 0.5
        ignore_thresholds: false
        negatives_lower_than_unmatched: true
        force_match_for_each_row: true
      }
    }
    similarity_calculator {
      iou_similarity {
      }
    }
    anchor_generator {
      ssd_anchor_generator {
        num_layers: 6
        min_scale: 0.2
        max_scale: 0.95
        aspect_ratios: 1.0
        aspect_ratios: 2.0
        aspect_ratios: 0.5
        aspect_ratios: 3.0
        aspect_ratios: 0.3333
      }
    }
    image_resizer {
      fixed_shape_resizer {
        height: 300
        width: 300
      }
    }
    box_predictor {
      convolutional_box_predictor {
        min_depth: 0
        max_depth: 0
        num_layers_before_predictor: 0
        use_dropout: false
        dropout_keep_probability: 0.8
        kernel_size: 1
        box_code_size: 4
        apply_sigmoid_to_scores: false
        conv_hyperparams {
          activation: RELU_6,
          regularizer {
            l2_regularizer {
              weight: 0.00004
            }
          }
          initializer {
            truncated_normal_initializer {
              stddev: 0.03
              mean: 0.0
            }
          }
          batch_norm {
            train: true,
            scale: true,
            center: true,
            decay: 0.9997,
            epsilon: 0.001,
          }
        }
      }
    }
    feature_extractor {
      type: 'ssd_mobilenet_v1'
      min_depth: 16
      depth_multiplier: 1.0
      conv_hyperparams {
        activation: RELU_6,
        regularizer {
          l2_regularizer {
            weight: 0.00004
          }
        }
        initializer {
          truncated_normal_initializer {
            stddev: 0.03
            mean: 0.0
          }
        }
        batch_norm {
          train: true,
          scale: true,
          center: true,
          decay: 0.9997,
          epsilon: 0.001,
        }
      }
    }
    loss {
      classification_loss {
        weighted_sigmoid {
        }
      }
      localization_loss {
        weighted_smooth_l1 {
        }
      }
      hard_example_miner {
        num_hard_examples: 3000
        iou_threshold: 0.99
        loss_type: CLASSIFICATION
        max_negatives_per_positive: 3
        min_negatives_per_image: 0
      }
      classification_weight: 1.0
      localization_weight: 1.0
    }
    normalize_loss_by_num_matches: true
    post_processing {
      batch_non_max_suppression {
        score_threshold: 1e-8
        iou_threshold: 0.6
        max_detections_per_class: 100
        max_total_detections: 100
      }
      score_converter: SIGMOID
    }
  }
}

train_config: {
  batch_size: 30
  optimizer {
    rms_prop_optimizer: {
      learning_rate: {
        exponential_decay_learning_rate {
          initial_learning_rate: 0.004
          schedule {
            step: 800000
            learning_rate: 0.004
          }
          schedule {
            step: 1000000
            learning_rate: 0.002
          }
          decay_steps: 800720
          decay_factor: 0.95
        }
      }
      momentum_optimizer_value: 0.9
      decay: 0.9
      epsilon: 1.0
    }
  }
  fine_tune_checkpoint: "/home/a/yxf/ssd_mobile_train/ssd_mobilenet_v1_coco/model.ckpt"
  from_detection_checkpoint: true

  num_steps: 1000000
  data_augmentation_options {
    random_horizontal_flip {
    }
  }
  data_augmentation_options {
    ssd_random_crop {
    }
  }
}

train_input_reader: {
  tf_record_input_reader {
    input_path: "path/data/record/train.record"
  }
  label_map_path: "path/object_label_map.pbtxt"
}

eval_config: {
  num_examples:   8000
  max_evals: 10
}

eval_input_reader: {
  tf_record_input_reader {
    input_path: "path/data/record/test.record"
  }
  label_map_path: "path/object_label_map.pbtxt"
  shuffle: false
  num_readers: 1
}

`

### this is the right result !
![ok1](https://user-images.githubusercontent.com/32695021/70208040-7d54c800-1767-11ea-92fc-3e2a3df0abfd.jpg)
![ok2](https://user-images.githubusercontent.com/32695021/70208042-7ded5e80-1767-11ea-9c3f-f1be75028258.jpg)
### but  this is err  result !
![err](https://user-images.githubusercontent.com/32695021/70208039-7d54c800-1767-11ea-84ec-34f26d3a9a5d.jpg)
![err2](https://user-images.githubusercontent.com/32695021/70208307-28fe1800-1768-11ea-9432-cd3cad6ab087.jpg)

**Comment by alanjuster** [comment]
前两张图片显示预测结果是 螺丝刀（物体就是螺丝刀），后两张预测结果是螺丝刀和管钳（但物体是显示器和键盘，这是错误的），这就是现在问题，还请教下怎么才能解决这个问题！

The first two pictures show that the predicted result is a screwdriver (the object is a screwdriver), and the last two predictions are a screwdriver and a pipe wrench (but the object is a monitor and keyboard, which is wrong). This is the problem now, and how to solve this problem!

**Comment by alanjuster** [comment]
我找到问题所在了，处理后识别就绕开了这个干扰,  需要做的就是把coco数据集作为背景数据去重新训练，就是这样，具体怎么做可以网上搜索，还不明白的话，可以在这留言。

now  i  find  what 's  my problem， and  now  my train  model  is  more better than before,   you need set all coco_dataset as  you trainning  background,  that's it

**Comment by SLAMgreen** [comment]
> 我找到问题所在了，处理后识别就绕开了这个干扰, 需要做的就是把coco数据集作为背景数据去重新训练，就是这样，具体怎么做可以网上搜索，还不明白的话，可以在这留言。
> 
> now i find what 's my problem， and now my train model is more better than before, you need set all coco_dataset as you trainning background, that's it

请问，怎么将COCO数据集作为背景数据呢？因为预训练模型不是已经基于COCO训练这个模型了吗，如果再基于COCO训练意义是什么呢？
先感谢大佬解答。

**Comment by alanjuster** [comment]
不好意思，快半个月了 才看到！
我的理解就是使用的是coco预训练模型的参数信息而已，并没有把coco的训练完成的权重信息加入，至于如何把数据集处理成背景问题：
一、批量对coco数据集遍历 生成对应xml（格式和正常样本打标签的xml一样），但xml 不可使用 训练样本的label，因为是背景不是物体，
       随便给个别的label就行了，宽高涵盖就是图片宽高 ；
二、把背景的xml集和标注样本的xml集混合打乱（不然训练曲线就是心跳一样的过山车，不稳定）
三、正常的 xml--> csv    csv-->tfrecoreds(有个获取labels的方法，在里面有label=‘**’，return 1  label =‘**’ return 2   else  return  0) 
       这个 0就是做背景处理了，也就是只要labels 不是你要训练的，就是默认做背景处理。  
四、其余跟正常训练一样

**Comment by SLAMgreen** [comment]
谢谢回答。



---Original---
